apps/prairielearn/src/question-servers/calculation-subprocess.js

```diff
@@ -105,21 +95,15 @@ async function callFunction(func, question_course, question, inputData) {
   }
 }
 
-module.exports.generate = (question, course, variant_seed, callback) => {
-  callFunction('generate', course, question, { variant_seed }).then(
-    ({ data, courseIssues }) => callback(null, courseIssues, data),
-    (err) => callback(err),
-  );
-};
+export async function generate(question, course, variant_seed) {
+  return await callFunction('generate', course, question, { variant_seed });
+}
 
-module.exports.grade = (submission, variant, question, question_course, callback) => {
-  callFunction('grade', question_course, question, { submission, variant }).then(
-    ({ data, courseIssues }) => callback(null, courseIssues, data),
-    (err) => callback(err),
-  );
-};
+export async function grade(submission, variant, question, question_course) {
+  return await callFunction('grade', question_course, question, { submission, variant });
+}
 
-module.exports.getFile = (filename, variant, question, course, callback) => {
+export function getFile(filename, variant, question, course, callback) {
```

Comment:
I'm curious about this function. It doesn't seem to be called from anywhere in the code, but since my understanding of v2 questions is somewhat limited I'd prefer not to remove it without someone else checking it out. Note that this function may be an invalid implementation of `file`, used in freeform questions with dynamic file generation, since the names don't match but the arguments do. Is the concept of dynamic file generation in v2 questions available, and if so, should this function be renamed to `file`?

Reply:
Huh... I went spelunking and it _seems_ like v2 questions are supposed to support dynamic files, through I agree that it seems like there's not really any way this code would work as written. @mwest1066 tagging you in again in case you know 😬

Reply:
I found 7 questions matching `org:PrairieLearn getFile path:server.js` with GitHub search, all from `pl-ae353`. All of these questions had QIDs of the form `deprecated_<XXX>` and are not used in any assessments in that course. So it looks like @nwalters512 is correct and this was once supported in v2 questions. I guess it broke at some point and no one noticed because it hasn't been used by anyone in a long time. Looks like it's safe to remove.

Reply:
Removed in bb0147b58.